otx/algorithms/classification/tasks/inference.py

```diff
@@ -325,6 +328,9 @@ def _init_model_cfg(self):
             cfg_path = os.path.join(base_dir, "model_hierarchical.py")
         else:
             cfg_path = os.path.join(base_dir, "model.py")
+        # Override the model configuration if we are in SELFSUPERVISED mode
+        if self._hyperparams.algo_backend.train_type == TrainType.SELFSUPERVISED:
```

Comment:
This if could be the outer one:
```
if self._hyperparams.algo_backend.train_type == TrainType.SELFSUPERVISED:
   ...
else:
   base_dir = os.path.abspath(os.path.dirname(self.template_file_path))
   ...
```
Such structure is a bit more clear to extend in the future.